### PR TITLE
[IMP] iot: Filter Printers Based On Subtype

### PR DESCRIPTION
**Pull request by viforget**
Add a field in the devices models for the category of the printer The subtype will automatically be selected via the device identifier

task id : 2146229

### CODE REVIEW

File: addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py [hunk 1]
```diff
@@ -86,6 +86,15 @@ def __init__(self, identifier, device):
...
         self.receipt_protocol = 'star' if 'STR_T' in device['device-id'] else 'escpos'
+
+        if 'CMD:ESC/POS;' in device['device-id']:
+            self.device_subtype = "receipt_printer"
+        elif 'CMD:STAR;' in device['device-id']:
+            self.device_subtype = "star_printer"
+        elif "COMMAND SET:ZPL;" in device['device-id']:
+            self.device_subtype = "label_printer"
+        else:
+            self.device_subtype = "office_printer"
```
Comment (Ysoroko) [inline addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py:97]:
```python
printer_subtypes = {
    'CMD:ESC/POS;' : 'receipt_printer',
    'CMD:STAR;' : 'star_printer',
    'COMMAND SET:ZPL;' : 'label_printer',
}

self.device_subtype = printer_subtypes.get(device['device-id'], 'office_printer')
```

**Review by Ysoroko — CHANGES_REQUESTED** [review]

**Comment by viforget** [comment]
@Ysoroko

File: addons/hw_drivers/iot_handlers/drivers/PrinterDriver_W.py [hunk 2]
```diff
@@ -55,6 +55,12 @@ def __init__(self, identifier, device):
         })
 
         self.receipt_protocol = 'escpos'
+        if any(cmd in device['iddentifer'] for cmd in ['STAR', 'Receipt']):#Find the correct STR to find
```
Comment (Ysoroko) [inline addons/hw_drivers/iot_handlers/drivers/PrinterDriver_W.py:58]: typo in "iddentifier"

Comment (Ysoroko) [inline addons/hw_drivers/iot_handlers/drivers/PrinterDriver_W.py:58]: Remove the comment, it doesn't add any information

File: addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py [hunk 3]
```diff
@@ -86,6 +86,13 @@ def __init__(self, identifier, device):
         })
 
         self.receipt_protocol = 'star' if 'STR_T' in device['device-id'] else 'escpos'
+
+        if any(cmd in device['device-id'] for cmd in ['CMD:STAR;', 'CMD:ESC/POS;']):
+            self.device_subtype = "receipt_printer"
+        elif "COMMAND SET:ZPL;" in device['device-id']:
+            self.device_subtype = "label_printer"
+        else:
+            self.device_subtype = "office_printer"
```
Comment (Ysoroko) [inline addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py:95]: Same as below for windows, "office_printer" doesn't correspond to the "other_printer" defined in your model "subtype" selection field

File: addons/hw_drivers/iot_handlers/drivers/PrinterDriver_W.py [hunk 4]
```diff
@@ -55,6 +55,12 @@ def __init__(self, identifier, device):
...
+            self.device_subtype = "receipt_printer"
+        elif "ZPL" in device['identifier']:
+            self.device_subtype = "label_printer"
+        else:
+            self.device_subtype = "office_printer"
```
Comment (Ysoroko) [inline addons/hw_drivers/iot_handlers/drivers/PrinterDriver_W.py:63]:
```Python
self.device_subtype = "office_printer"
```
This doesn't correspond to the model's "other_printer" which itself corresponds to "Office Printer".

**Review by Ysoroko — CHANGES_REQUESTED** [review]

**Comment by Ysoroko** [comment]
@robodoo r+

**Pull request closed without merging**